File: lib/app/Models/CalculateTariffsParametersDTO.php
<?php
/**
 * CalculateTariffsParametersDTO
 */
namespace app\Models;

/**
 * CalculateTariffsParametersDTO
 * @description Параметры для расчета стоимости услуг.
 */
class CalculateTariffsParametersDTO {

    /** @var int $campaignId Идентификатор кампании. У пользователя, который выполняет запрос, должен быть доступ к этой кампании.  Используйте параметр &#x60;campaignId&#x60;, если уже завершили подключение магазина на Маркете. Иначе вернется пустой список.  Обязательный параметр, если не указан параметр &#x60;sellingProgram&#x60;. Совместное использование параметров приведет к ошибке.*/
    public $campaignId = 0;

    /** @var string $sellingProgram */
    public $sellingProgram = "";

    /** @var string $frequency */
    public $frequency = "";

}

File: lib/Models/CalculateTariffsParametersDTO.cpp


#include "CalculateTariffsParametersDTO.h"

using namespace Tiny;

CalculateTariffsParametersDTO::CalculateTariffsParametersDTO()
{
	campaignId = long(0);
	sellingProgram = SellingProgramType();
	frequency = PaymentFrequencyType();
}

CalculateTariffsParametersDTO::CalculateTariffsParametersDTO(std::string jsonString)
{
	this->fromJson(jsonString);
}

CalculateTariffsParametersDTO::~CalculateTariffsParametersDTO()
{

}

void
CalculateTariffsParametersDTO::fromJson(std::string jsonObj)
{
    bourne::json object = bourne::json::parse(jsonObj);

    const char *campaignIdKey = "campaignId";

    if(object.has_key(campaignIdKey))
    {
        bourne::json value = object[campaignIdKey];



        jsonToValue(&campaignId, value, "long");


    }

    const char *sellingProgramKey = "sellingProgram";

    if(object.has_key(sellingProgramKey))
    {
        bourne::json value = object[sellingProgramKey];




        SellingProgramType* obj = &sellingProgram;
		obj->fromJson(value.dump());

    }

    const char *frequencyKey = "frequency";

    if(object.has_key(frequencyKey))
    {
        bourne::json value = object[frequencyKey];




        PaymentFrequencyType* obj = &frequency;
		obj->fromJson(value.dump());

    }


}

bourne::json
CalculateTariffsParametersDTO::toJson()
{
    bourne::json object = bourne::json::object();





    object["campaignId"] = getCampaignId();







	object["sellingProgram"] = getSellingProgram().toJson();






	object["frequency"] = getFrequency().toJson();


    return object;

}

long
CalculateTariffsParametersDTO::getCampaignId()
{
	return campaignId;
}

void
CalculateTariffsParametersDTO::setCampaignId(long  campaignId)
{
	this->campaignId = campaignId;
}

SellingProgramType
CalculateTariffsParametersDTO::getSellingProgram()
{
	return sellingProgram;
}

void
CalculateTariffsParametersDTO::setSellingProgram(SellingProgramType  sellingProgram)
{
	this->sellingProgram = sellingProgram;
}

PaymentFrequencyType
CalculateTariffsParametersDTO::getFrequency()
{
	return frequency;
}

void
CalculateTariffsParametersDTO::setFrequency(PaymentFrequencyType  frequency)
{
	this->frequency = frequency;
}



